CMakeLists.txt

```diff
@@ -1,20 +1,56 @@
 cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
-project(vHLLE_SMASH_particle_sampler)
+project(SMASH-hadron-sampler VERSION "3.1.1" LANGUAGES CXX)
+set(SAMPLER_VERSION "${SMASH-hadron-sampler_VERSION}")
+# If the following set() command is not commented out, the codebase is marked as
+# dirty (for development on the way to next release)
+# set(SAMPLER_VERSION "${SAMPLER_VERSION}-next")
+
 include(CTest)
 
 # Tell cmake where to find modules
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
 
 
-# Determine System and set the relevant generic compiler flags accordingly (optimisation + warnings) 
+# Determine System and set the relevant generic compiler flags accordingly (optimisation + warnings)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -W -Wall -Wextra -Wmissing-declarations -Wpointer-arith -Wshadow -Wuninitialized -Winit-self -Wundef -Wcast-align -Wformat=2 -Wold-style-cast -Werror=switch -std=c++17")
 if(NOT (APPLE AND
    CMAKE_HOST_SYSTEM_PROCESSOR MATCHES "^arm" AND
    CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND
    CMAKE_CXX_COMPILER_VERSION VERSION_LESS "15"))
-     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")  
+     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+endif()
+message(STATUS "CXX_FLAGS = " ${CMAKE_CXX_FLAGS})
+
+# ------------------------------------------------------------------------------
+# Retrieve all metadata for --version option
+# ------------------------------------------------------------------------------
+# The version number is already set via the project() command, since a VERSION option is provided
+# in it, and passed to the variable SAMPLER_VERSION via the set() command after it. The following
+# checks if .git information about the branch and the commit hash is available.
```

Comment:
This comment should be moved before `git_describe` call. Here, the code that follows extract information setting `GIT_BRANCH`.